### PR TITLE
[interpreter] Fix JS generator

### PR DESCRIPTION
**Pull request by rossberg**
Fix bugs reported by @gahaas.

### CODE REVIEW

File: interpreter/script/js.ml [hunk 1]
```diff
@@ -347,6 +346,8 @@ let wrap item_name wrap_action wrap_assertion at =
       {module_name = Utf8.decode "spectest"; item_name = Utf8.decode "eq_ref";
        idesc = FuncImport (4l @@ at) @@ at} @@ at ]
   in
+  let idesc_is_func = match idesc.it with FuncImport _ -> 1l | _ -> 0l in
```
Comment (gahaas) [inline interpreter/script/js.ml]: I don't completely understand the logic here, but for me it causes an off-by-one error. Just using
`let item = Lib.List32.length imports @@ at in` in the next line fixes the problem for me.

File: interpreter/script/js.ml [hunk 2]
```diff
@@ -83,7 +83,7 @@ let harness =
   "}\n" ^
   "\n" ^
   "function exports(instance) {\n" ^
-  "  return {module: instance.exports, host: {ref: hostref}};\n" ^
+  "  return {module: instance.exports, spectest: spectest};\n" ^
```
Comment (gahaas) [inline interpreter/script/js.ml]: The generated module expects an import `spectest::eq_ref`. Could you extend `spectest` above with an `eq_ref` field?

Comment (gahaas) [reply interpreter/script/js.ml]: Also, some tests expect that `hostref(1)` provide an exported wasm function, but it doesn't.

Comment (rossberg) [reply interpreter/script/js.ml]: Added missing eq_ref. I'm not sure I understand your other comment. Which test is this?

File: interpreter/script/js.ml (hunk 1 above)
Comment (rossberg) [reply interpreter/script/js.ml]: Oops, that went the wrong way round. Replaced with a correct and less brittle version.

File: interpreter/script/js.ml (hunk 2 above)
Comment (gahaas) [reply interpreter/script/js.ml]: For the test in `ref_is_null.wast:33`, `hostref(1)` is passed in as a valid value of funcref. However, afaict, it is not a valid value. When I change `hostref(1)` with `$1.exports["funcref-elem"]`, which is a valid funcref value, the test passes.

Comment (rossberg) [reply interpreter/script/js.ml]: Ah, thanks! That test actually was bogus (but the interpreter does not type-check script commands like `invoke` -- it probably should). I removed it.

**Pull request merged**